x/delayedack/keeper/invariants_test.go

```diff
@@ -0,0 +1,312 @@
+package keeper_test
+
+import (
+	"github.com/tendermint/tendermint/libs/rand"
+
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+	"github.com/dymensionxyz/dymension/v3/x/delayedack/keeper"
+	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+	rollapptypes "github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+func (suite *DelayedAckTestSuite) TestInvariants() {
+	suite.SetupTest()
+	initialheight := int64(10)
+	suite.Ctx = suite.Ctx.WithBlockHeight(initialheight)
+
+	numOfRollapps := 10
+	numOfStates := 10
+	// create rollapps
+	seqPerRollapp := make(map[string]string)
+	rollappBlocks := make(map[string]uint64)
+
+	for i := 0; i < numOfRollapps; i++ {
+		rollapp := suite.CreateDefaultRollapp()
+		seqaddr := suite.CreateDefaultSequencer(suite.Ctx, rollapp)
+
+		// skip one of the rollapps so it won't have any state updates
+		if i == 0 {
+			continue
+		}
+		seqPerRollapp[rollapp] = seqaddr
+		rollappBlocks[rollapp] = 0
+
+	}
+
+	sequence := uint64(0)
+	for j := 0; j < numOfStates; j++ {
+
+		numOfBlocks := uint64(rand.Intn(10) + 1)
+		for rollapp, sequencer := range seqPerRollapp {
+
+			_, err := suite.PostStateUpdate(suite.Ctx, rollapp, sequencer, rollappBlocks[rollapp]+uint64(1), numOfBlocks)
+			suite.Require().NoError(err)
+			for k := uint64(1); k <= numOfBlocks; k++ {
+				// calculating a different proof height incrementing a block height for each new packet
+				proofheight := rollappBlocks[rollapp] + k
+				rollappPacket := &commontypes.RollappPacket{
+					RollappId:   rollapp,
+					Packet:      getNewTestPacket(sequence),
+					Status:      commontypes.Status_PENDING,
+					ProofHeight: proofheight,
+				}
+				err := suite.App.DelayedAckKeeper.SetRollappPacket(suite.Ctx, *rollappPacket)
+				suite.Require().NoError(err)
+
+				sequence++
+			}
+			rollappBlocks[rollapp] = rollappBlocks[rollapp] + numOfBlocks
+		}
+
+		suite.Ctx = suite.Ctx.WithBlockHeight(suite.Ctx.BlockHeader().Height + 1)
+	}
+
+	// progress finalization queue
+	err := suite.App.RollappKeeper.FinalizeQueue(suite.Ctx)
+	suite.Require().NoError(err)
+
+	// test fraud
+	for rollapp := range seqPerRollapp {
+		err := suite.App.DelayedAckKeeper.HandleFraud(suite.Ctx, rollapp)
+		suite.Require().NoError(err)
+		break
+	}
+
+	// check invariant
+	msg, bool := keeper.AllInvariants(suite.App.DelayedAckKeeper)(suite.Ctx)
+	suite.Require().False(bool, msg)
```

Comment:
```suggestion
	msg, fails := keeper.AllInvariants(suite.App.DelayedAckKeeper)(suite.Ctx)
	suite.Require().False(bool, fails)
```
bool collides

Reply:
i guess you meant to replace bool by fails